testsuite/MDAnalysisTests/analysis/test_base.py

```diff
@@ -162,6 +162,25 @@ def test_AnalysisFromFunction(u, start, stop, step, nframes):
         assert_equal(results, ana.results)
 
 
+def test_AnalysisFromFunction_args_order(u):
+    # Issue 2503
+    def mass_xyz(atomgroup, masses):
+        return atomgroup.positions * masses
+    
+    protein = u.select_atoms('protein')
+    sub = u.atoms[:100]
+    #ans1 = base.AnalysisFromFunction(mass_xyz, u.trajectory, protein, protein.masses.reshape(-1, 1))
+    # lack traj but can find in protein
+    ans2 = base.AnalysisFromFunction(mass_xyz, protein, protein.masses.reshape(-1, 1))
+    # multiple traj input
+    ans3 = base.AnalysisFromFunction(mass_xyz, protein, sub, protein.masses.reshape(-1, 1))
+    
+    assert len(ans2.args) == 2
+    assert (ans2.args[0] == protein) and not (ans2.args[1] - protein.masses.reshape(-1, 1)).any()
+    assert ans2._trajectory == protein.universe.trajectory
+    assert ans3.args[0] == protein and ans3.args[1] == sub
+   
```

Comment:
Inspecting `ans` is good. (And  multiple `assert` for the _same_ `ans` are fine here.)

I would also add a short `ans.run(stop=5)` to make sure it really runs for a few steps. You should get something in `ans.result`. Just do a mean or a sum over whatever values you find there and check that value as an additional test.